src/navigation/checkout/PaymentMethodEdenred.js

```diff
@@ -0,0 +1,124 @@
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    if (cart.hasEdenredCredentials) {
+      loadPaymentDetails()
+    } else {
+      authorize({
+        clientId: clientId,
+        redirectUrl: `${Config.APP_AUTH_REDIRECT_SCHEME}://edenred`,
+        serviceConfiguration: {
+          authorizationEndpoint: `${authorizationEndpoint}/connect/authorize`,
+          tokenEndpoint: `${baseURL}/edenred/connect/token`,
+        },
+        additionalParameters: {
+          acr_values: 'tenant:fr-ctrtku',
+          nonce: uuidv4(),
+          ui_locales: 'fr-FR',
+        },
+        scopes: ['openid', 'edg-xp-mealdelivery-api', 'offline_access'],
+        dangerouslyAllowInsecureHttpRequests: __DEV__,
+        useNonce: false,
+        usePKCE: false,
+      }).then(result => {
+        updateEdenredCredentials(result.accessToken, result.refreshToken);
+      }).catch(error => {
+        console.log(error)
+        navigation.navigate('CheckoutPaymentMethodCard');
+      });
+    }
+  }, [ baseURL, cart.hasEdenredCredentials, clientId, loadPaymentDetails, navigation, updateEdenredCredentials, authorizationEndpoint ]);
+
+  if (paymentDetailsLoaded && paymentDetails.payments.length === 2) {
+
+    const cardPayment    = _.find(paymentDetails.payments, p => p.method.code === 'CARD');
+    const edenredPayment = _.find(paymentDetails.payments, p => p.method.code === 'EDENRED')
+
+    return (
+      <HeaderHeightAwareKeyboardAvoidingView>
+        <VStack p="2" flex={1} justifyContent="space-between">
+          <VStack>
+            <HStack justifyContent="space-between">
+              <Text fontSize="md">{ t('TOTAL') }</Text>
+              <Text fontSize="md">{ formatPrice(cart.total) }</Text>
+            </HStack>
+            <HStack justifyContent="space-between">
+              <Text fontSize="md" fontWeight="bold">{ t('EDENRED_ELIGIBLE_AMOUNT') }</Text>
+              <Text fontSize="md" fontWeight="bold">{ formatPrice(edenredPayment.amount) }</Text>
+            </HStack>
+            <HStack justifyContent="space-between">
+              <Text fontSize="md">{ t('EDENRED_COMPLEMENT') }</Text>
+              <Text fontSize="md">{ formatPrice(cardPayment.amount) }</Text>
+            </HStack>
+          </VStack>
+          <CreditCard
+            cart={ cart }
+            errors={ errors }
+            shouldLoadPaymentDetails={ false }
+            total={ cardPayment.amount }
+            onSubmit={ (values) => {
+              const { cardholderName, savedCardSelected, saveCard } = values;
+              checkout(cardholderName, savedCardSelected, saveCard);
+            }}
+          />
+        </VStack>
+      </HeaderHeightAwareKeyboardAvoidingView>
+    )
+  }
+
+  return (
+    <VStack flex={ 1 } p="2" justifyContent="space-between">
+      <Skeleton.Text />
+      <HStack p="3">
+        <Skeleton startColor="primary.300" />
+      </HStack>
+    </VStack>
+  )
+}
+
+function mapStateToProps(state) {
```

Comment:
Maybe we should discuss it first, but for me, using `useSelector` and `useDispatch` makes code easier to understand, rather than scrolling back and forth to know what each prop means. See, for example:  https://github.com/coopcycle/coopcycle-app/blob/645241035b8e9bd96eb481a3b483aee1f44f2280/src/navigation/account/RegisterConfirm.js